### PR TITLE
[Settings] As the instance is created in the constructor we don't nee…

### PR DESCRIPTION
**Pull request by TheOneRing**
…d to check for it

### CODE REVIEW

File: src/gui/owncloudgui.cpp [hunk 1]
```diff
@@ -1025,8 +1020,7 @@ void ownCloudGui::slotShutdown()
     // that saving the geometries happens ASAP during a OS shutdown
 
     // those do delete on close
-    if (!_settingsDialog.isNull())
-        _settingsDialog->close();
+    _settingsDialog->close();
```
Comment (jnweiger) [inline src/gui/owncloudgui.cpp]: Is it safe to ->close()M if .isNull() ?

File: src/gui/settingsdialog.cpp [hunk 2]
```diff
@@ -84,13 +84,11 @@ QString shortDisplayNameForSettings(OCC::Account *account, int width)
 namespace OCC {
 
 SettingsDialog::SettingsDialog(ownCloudGui *gui, QWidget *parent)
-    : QDialog(parent)
+    : QDialog(parent, Qt::WindowFlags() & ~Qt::WindowContextHelpButtonHint)
     , _ui(new Ui::SettingsDialog)
```
Comment (jnweiger) [inline src/gui/settingsdialog.cpp]: Should we also setAttribute(Qt::WA_DeleteOnClose, true); here?

Comment (TheOneRing) [reply src/gui/settingsdialog.cpp]: The widget is only initialised once so we don't want it to be deleted.

File: src/gui/owncloudgui.cpp (hunk 1 above)
Comment (TheOneRing) [reply src/gui/owncloudgui.cpp]: The dialogue is initialised in the constructor and should always exist.

File: src/gui/settingsdialog.cpp (hunk 2 above)
Comment (ogoffart) [reply src/gui/settingsdialog.cpp]: But don't we want to delete it when we exit?

File: src/gui/owncloudgui.cpp [hunk 3]
```diff
@@ -97,12 +94,17 @@ ownCloudGui::ownCloudGui(Application *parent)
         this, &ownCloudGui::slotShowGuiMessage);
 }
 
+ownCloudGui::~ownCloudGui()
+{
+    _settingsDialog->deleteLater();
+}
```
Comment (dschmidt) [inline src/gui/owncloudgui.cpp]: Why do we need this? Isn't the SettingsDialog parented to the ownCloudGui anyway?

Comment (TheOneRing) [reply src/gui/owncloudgui.cpp]: ownCloudGui is a QObject and the dialog a Widget so it doesn't work here.

**Comment by TheOneRing** [comment]
> I don't think the title/description of the PR is accurate.
> 
> It should be more around the lines of:
> 
> > Create the settings dialog only once and don't delete it when it's hidden
> > This allows us to simplify the code
> 
> I'm not sure whether this is a good or a bad thing. 🤷‍♂

We had quite some code to recreate the dialog in case it was deleted.
But the current code never deleted it so it was just dead code.

> 
> Before approving I would like to know why we now need `_settingsDialog->deleteLater()`.

File: src/gui/owncloudgui.cpp (hunk 3 above)
Comment (dschmidt) [reply src/gui/owncloudgui.cpp]: Huh? Can you elaborate a bit more?

**Comment by TheOneRing** [comment]
ping

File: src/gui/systray.h [hunk 4]
```diff
@@ -34,6 +34,7 @@ class Systray : public QSystemTrayIcon
 {
     Q_OBJECT
 public:
+    Systray(QObject *parent = nullptr);
```
Comment (ogoffart) [inline src/gui/systray.h]: Btw, one can do `using QSystemTrayIcon::QSystemTrayIcon;`  to inherit the constructor.

**Pull request merged**